aqueduct/flow.py

```diff
@@ -411,13 +412,20 @@ async def _check_is_alive(self, sleep_sec: float = 1.):
         If at least one process is not alive, it stops Flow.
         """
         while self.state != FlowState.STOPPED:
+            processes_stats = ProcessesStats()
             for handler, context in self._contexts.items():
                 for proc in context.processes:
                     if not proc.is_alive():
                         if self.is_running:
+                            handler_name = handler.__class__.__name__
                             log.error('The process %s for %s handler is dead',
-                                      proc.pid, handler.__class__.__name__)
+                                      proc.pid, handler_name)
+                            processes_stats.add_dead_process()
+                            self._metrics_manager.collector.add_processes_stats(processes_stats)
                             await self.stop(graceful=False)
+                    else:
+                        processes_stats.add_running_process()
+            self._metrics_manager.collector.add_processes_stats(processes_stats)
```

Comment:
Look like ProcessesStats() continiuosly increments,  i think you need to reset to zero sometimes